barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.cpp

```diff
@@ -370,7 +377,46 @@ void process_honk_recursion_constraints(Builder& builder,
         // final recursion output.
         builder.set_recursive_proof(current_aggregation_object);
     }
-};
+}
+
+// TODO(https://github.com/AztecProtocol/barretenberg/issues/1095): Probably makes sense to aggregate Honk and AVM
+// proofs together.
+#ifndef DISABLE_AZTEC_VM
+void process_avm_recursion_constraints(Builder& builder,
+                                       AcirFormat& constraint_system,
+                                       bool has_valid_witness_assignments,
+                                       GateCounter<Builder>& gate_counter)
+{
+    AggregationObjectIndices current_aggregation_object =
+        stdlib::recursion::init_default_agg_obj_indices<Builder>(builder);
+
+    // Add recursion constraints
+    size_t idx = 0;
+    for (auto& constraint : constraint_system.avm_recursion_constraints) {
+        current_aggregation_object = create_avm_recursion_constraints(
+            builder, constraint, current_aggregation_object, has_valid_witness_assignments);
+
+        gate_counter.track_diff(constraint_system.gates_per_opcode,
+                                constraint_system.original_opcode_indices.avm_recursion_constraints.at(idx++));
+    }
+
+    // Now that the circuit has been completely built, we add the output aggregation as public
+    // inputs.
+    if (!constraint_system.avm_recursion_constraints.empty()) {
```

Comment:
if we ever try to create a circuit with both honk and avm recursion constraints, this will throw an error, but I think we can resolves this later.

Reply:
error comes from calling set_recursive_proof multiple times.

Reply:
I suggest that I add a comment and a reference to the issue 1095 to solve this. @lucasxia01 Ok?

Reply:
sounds fine